### DISCUSSION TITLE
Rate limited when using continuationToken

### DISCUSSION THREAD

**Discussion by joshr4** [skymethod/op3#39] [Q&A]
I have a script to fetch the last 30 day's worth of data for a show, and i have the limit set to `limit=1000`, and am getting back results but wind up getting a `429` response code back from the op3 api.

Do I need to lower my limit to reduce the number of requests or try and space out the requests myself in order to fall within the rate limits? Are op3 API rate limits defined anywhere?

here is the url I'm using:

https://op3.dev/api/1/downloads/show/8f110b538260417fa806835df8e0fddb?start=2024-11-20&end=2024-12-20T00:00:00.000Z&format=json&limit=1000

**Comment by johnspurlock-skymethod** [comment]
Hi there, yes the OP3 api is currently rate-limited after being swamped with requests a few months ago.  Since OP3 is open source, you can find the spot in the code where it's done!

It is call based, not parameter based, so feel free to use any `limit` param, but keep your calls to average about one a second if you are getting those `429 slow down` responses
